### DISCUSSION TITLE
Custom script timeout

### DISCUSSION THREAD

**Discussion by davidgib01** [netbox-community/netbox#14738] [Help Wanted!]
Hello,
I am using version 3.6.2 of netbox and I have set up custom GUI scripts to scan my network. The problem is that when the areas I scan are too large, I get a timout error after 5 minutes.
I tried to do as in a classic script and put at the beginning of my script:

```
def timeout_handler(signum, frame):
    raise TimeoutError("Le script a dépassé le temps imparti")

signal.signal(signal.SIGALRM, timeout_handler)
signal.alarm(10800)  # Définit la minuterie
```

and at the end:

`signal.alarm(0)`

Do any of you have a solution?

Sincerely

**Comment by candlerb** [comment]
See:
* https://docs.netbox.dev/en/stable/customization/custom-scripts/#job_timeout
* https://docs.netbox.dev/en/stable/configuration/miscellaneous/#rq_default_timeout

**Comment by davidgib01** [comment]
Hello,

Thank you for your reply.

I really missed this information.

Thank you and happy new year :D
